tests/test_matrices_operations.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+from scipy.sparse import spdiags
+from scipy.sparse import rand as sprand
+from scipy.sparse.linalg import LinearOperator
+from scipy.sparse.linalg import cg
+from solvers.DPGD import get_inverse_D
+
+
+def test_inverse_D():
+    rng = np.random.RandomState(42)
+    p = rng.randint(2, 100)
+    data = np.array([-np.ones(p), np.ones(p)])
+    diags = np.array([0, 1])
+    D = spdiags(data, diags, p-1, p)
+    D_inv = get_inverse_D(p)
+    np.testing.assert_almost_equal(D @ D_inv, np.identity(p-1))
+
+
+def test_LO_solve_dual_eq():
```

Comment:
can you add a comment to explain what you are testing here?

Also, I was more into directly testing the code from `DPGD`, by calling it for 3 iterations with a small and dense `A` and then with a linear operator `LinearOperator(shape=A.shape, matvec=lambda x: A@x, ...)`. and checking that the results is the same.

I think you could actually to this for most solvers.